### PR TITLE
Improve Jenkinsfile

### PR DESCRIPTION
**Pull request by magicmatatjahu**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [ ] I have tested my changes.
- [ ] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:
- Repair failing generation of documentation

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -28,25 +28,35 @@ podTemplate(label: label) {
                         }
 
                         if(isMaster) {
-                            stage("prepare ssh for git") {
```
Comment (michal-hudy) [inline Jenkinsfile]: Why did you move it?

File: Jenkinsfile [hunk 2]
```diff
@@ -28,25 +28,35 @@ podTemplate(label: label) {
...
                                 stage("prepare docs and navigation") {
+                                    sh "./prepare-docs.sh -v ${params.DOCS_VERSION}"
+                                }
+
+                                stage("convert and copy manifest") {
+                                    execute("./manifest.sh -v ${params.DOCS_VERSION}")
+                                }
+
+                                stage("push new docs to master") {
                                     withCredentials([sshUserPrivateKey(credentialsId: "bitbucket-rw", keyFileVariable: 'sshfile')]) {
```
Comment (michal-hudy) [inline Jenkinsfile]: Is it necessary? Why you didn't use `ssh_key.pem` from `prepare ssh for git` step?

File: prepare-docs.sh [hunk 3]
```diff
@@ -5,45 +5,34 @@ GREEN='\033[0;32m'
...
-while getopts 's:v:' flag; do
+while getopts 'v:' flag; do
   case "${flag}" in
-    s) SSH_FILE=${OPTARG} ;;
     v) DOCS_VERSION=${OPTARG} ;;
   esac
 done
 
-# initial clean
-removeTemporaryFiles
```
Comment (michal-hudy) [inline prepare-docs.sh]: Why did you remove it? Script should always start with clean `temporaryFolder`

File: manifest.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+temporaryFolder="temporaryFolder"
```
Comment (michal-hudy) [inline manifest.sh]: You should verify if `temporaryFolder` exists and has necessary content.

Comment (magicmatatjahu) [reply manifest.sh]: If the content is not cloned, `content-to-json` will display a message that in `temporaryFolder` is empty, or `temporaryFolder` doesn't exist.

File: prepare-docs.sh (hunk 3 above)
Comment (michal-hudy) [reply prepare-docs.sh]: On Jenkins agree, but if someone will be testing it locally then `temporaryFolder` will need to be cleared manually.

File: manifest.sh (hunk 4 above)
Comment (michal-hudy) [reply manifest.sh]: Similar situation as for cleaning that folder before execution in `prepare-docs.sh` file.

File: commit-docs.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+# arguments
+SSH_FILE=
+DOCS_VERSION=
+
+# read arguments
+while getopts 's:v:' flag; do
+  case "${flag}" in
+    s) SSH_FILE=${OPTARG} ;;
+    v) DOCS_VERSION=${OPTARG} ;;
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply commit-docs.sh]: `SSH_FILE` and `DOCS_VERSION` for me is very descriptive flag names and I thinking that comments and descriptive alternatives are unnecessary.

File: Jenkinsfile (hunk 1 above)
Comment (magicmatatjahu) [reply Jenkinsfile]: I change logic, so this stage already is on this same place.

File: Jenkinsfile (hunk 2 above)
Comment (magicmatatjahu) [reply Jenkinsfile]: I changed logic, so yes, this is unnecessary.

File: prepare-docs.sh (hunk 3 above)
Comment (magicmatatjahu) [reply prepare-docs.sh]: Cloning will overwrite files in `temporaryFolder`. We can add `temporaryFolder` to `.gitignore`

File: commit-docs.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (michal-hudy) [reply commit-docs.sh]: You misunderstand me. You execute that script like that
```bash
./commit-docs.sh -s pathToKey -v someVersion
```
Where `-v` usually means `verbose`. So if you will change `-s` to `-ssh` and `-v` to `-version` it will be more descriptive...

File: prepare-docs.sh (hunk 3 above)
Comment (michal-hudy) [reply prepare-docs.sh]: Sure?
```bash
git clone https://github.com/kyma-project/kyma.git ./temporaryFolder
fatal: destination path './temporaryFolder' already exists and is not an empty directory.
```

File: scripts/convert-manifest.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+# read arguments
+while test $# -gt 0; do
+    case "$1" in
+        --docs-version | -v)
+            shift
+            DOCS_VERSION=$1
+            shift
+            ;;
+        --clone-repo-folder | -f)
```
Comment (michal-hudy) [inline scripts/convert-manifest.sh]: Path to `yaml` would be better, but ok :)
